### ISSUE TITLE
Duplicate tags appearing

### ISSUE THREAD

**Issue by ironhussar** [select2/select2#3951]
Hi,

I've just upgraded from S2 3.x to 4.x. Everything went pretty smoothly, apart from one thing. I have a list of items, that uses optgroups. Some items appear in more than one optgroup, so they have the same id. For example;

<select id="foo" class="form-control select2">

  <optgroup label="Group 1">
    <option value="19">green </option>
    <option value="18">red </option>
    <option value="14">test </option>
  </optgroup>

  <optgroup label="Group 2">
    <option value="11">Strategy </option>
    <option value="18">red </option>
  </optgroup>

  <optgroup label="Group 3">
    <option value="11">Strategy </option>
  </optgroup>

</select>

If I select Strategy in Group 2, the control automatically adds 2 tags, both called [Strategy]. If I close one using the X, only that one disappears. So the control is not using the value as the unique identifier for the tags.

Is there any way I can stop the duplicate tag appearing? It worked fine in 3.x, only showing one.

Thanks for any help.

**Comment by kevin-brown** [comment]
_Technically_, Select2 will consider two options with the same `id` to be distinct options, because they normally in a `<select>` box would be two separate options that just happened to share the same value.

----

Can you expand upon the issue you are facing by adding a few more pieces of information to your bug report?

- What steps can be used to reproduce the issue?
- What is the bug and what is the expected outcome?
- What browser(s) and Operating System have you tested with?
- Does the bug happen consistently across all tested browsers?
- What version of jQuery are you using? And what version of Select2?
- Are you using Select2 with other plugins?

The [contributing guide][cg] contains a useful section with a few questions which can help us track down the bug and speed up the process of finding a fix for it.  The creation of [an isolated test case][test-case] would also be useful.

[cg]: https://github.com/select2/select2/blob/master/CONTRIBUTING.md#reporting-bugs-with-select2
[test-case]: http://css-tricks.com/6263-reduced-test-cases/

**Comment by ironhussar** [comment]
Yep I can understand that. However pre 4.x it didn't exhibit this behaviour. As soon as I replaced my old Select2 with 4.x, it started happening.

Is there any reason why the same value element should show up twice (speaking from a design POV) -  if you have two elements with different names, they should also have different values. Or maybe it could be an option, not to render duplicate value items?

I can create a jsfiddle showing this if you like?

**Comment by ironhussar** [comment]
I am back to facing this issue again (temporarily I stopped grouping items to get around it).

Is there anything that I need to provide you with? Essentially the issue here is that when you are using optgroups any item that has the same ID will be rendered twice as a tag. The result when saving is fine (since it is the same ID), but on the front end, it doesn't look nice for the user to see a duplicate tag appearing in the select box.

**Comment by ironhussar** [comment]
I created a jsfiddle to show this issue: [https://jsfiddle.net/q3jh06px/2/](url)

Select the same named options from the list, and multiple tags appear in the box, with duplicate selected tags returned. Having duplicate ID numbers returned is not an issue, since that can be quickly filtered in processing code, but is it feasible to have an option to not render the same option twice?

This issue was not present in select2 3.x

**Comment by tapalilov** [comment]
I have the same issue. When select option which is dublicated in another options group add it twice. In 3.5.x when select value it disapear from the options.
